### DISCUSSION TITLE
Is `stats` omitting lines?

### DISCUSSION THREAD

**Discussion by SoongNoonien** [MusicPlayerDaemon/MPD#2036] [Q&A] [answered]
Hi!
Is it possible for the `stats` command to omit lines similar to `status` if a value is not known? Specifically is it possible that no line for `songs` is returned? I'm using this to decide if there any songs in the library when my client is connected.

**Comment by MaxKellermann** [comment]
I don't get it. If there are no songs in your library, then the value is known. It is zero.

**Reply by SoongNoonien** [reply]
Yes, this is true. But is there a possible state where the number of songs is not known and thus omitted in the output of `stats`?

**Reply by MaxKellermann** [reply]
I'm not sure, I can't think of any.

**Reply by SoongNoonien** [reply]
Hm, ok. Maybe if the database hasn't been created yet?

**Reply by MaxKellermann** [reply]
Then the database is empty, equals zero songs. If we'd change that, then MPD would need to suppress these lines whenever the database update is running.

**Reply by SoongNoonien** [reply]
I don't want to proposed that.

**Comment by SoongNoonien** [comment]
My concrete problem at the moment is this [issue](https://github.com/SoongNoonien/plattenalbum/issues/115). I'm trying to decide if it is my fault or a bug in mpd or its documentation. Unfortunately I'm not yet able to reproduce.

**Reply by MaxKellermann** [reply, marked as answer]
Oh, it is indeed possible for MPD to return empty stats. That can happen for example if the "proxy" database plugin fails; therefore you should really change your code to make it more robust against server-side errors where the response doesn't look like you expect it to be.
But in that user's setup, with the standard database plugin, that shouldn't happen. I can't explain what went wrong there, but in any case making code robust is a good idea.

**Reply by SoongNoonien** [reply]
Ok, thanks!
